src/Platform.c

```diff
@@ -204,12 +204,38 @@ void PltJoinThread(PLT_THREAD* thread) {
 #endif
 }
 
+void PltDetachThread(PLT_THREAD* thread)
+{
+    // Assume detached threads are no longer active
+    activeThreads--;
+
+#if defined(LC_WINDOWS)
+    // According MSDN:
+    // "Closing a thread handle does not terminate the associated thread or remove the thread object."
+    CloseHandle(thread->handle);
+#elif defined(__vita__)
+    sceKernelDeleteThread(thread->handle);
+#elif defined(__WIIU__)
+    OSDetachThread(&thread->thread);
+#elif defined(__3DS__)
+    threadDetach(thread->thread);
+#else
+    pthread_detach(thread->thread);
+#endif
+}
+
 void PltCloseThread(PLT_THREAD* thread) {
     activeThreads--;
 #if defined(LC_WINDOWS)
     CloseHandle(thread->handle);
 #elif defined(__vita__)
     sceKernelDeleteThread(thread->handle);
+#elif defined(__WIIU__)
+    // Thread is automatically closed after join
+#elif defined(__3DS__)
+    threadFree(thread->thread);
```

Comment:
PltJoinThread already calls threadFree(), so this will be a double-free.

Reply:
Oh good catch, I missed that.
I moved the `threadFree` call to `PltCloseThread` to be consistent with the other platforms.